src/test/java/net/minecraftforge/debug/ConfigTest.java

```diff
@@ -99,4 +133,48 @@ public SubCat(String value)
             }
         }
     }
+
+    public static class ConfigurationGui extends GuiConfig
```

Comment:
I'm currently looking into it, but it seems to require a breaking change. If I could replace the method IModGuiFactory.mainConfigGuiClass() with something that returns an already initialized GuiScreen, then this would be a good Idea. Currently you still need to implement an own class providing the arguments for the GuiConfig constructor, just so you can use IModGuiFactory.mainConfigGuiClass().